src/ops/key.rs

```diff
@@ -115,57 +113,69 @@ pub fn save_keys(
         }
     }
 
-    match diesel::insert_into(keys::table)
-        .values(keys_to_insert)
-        .execute(&mut db::establish_connection()?)
-    {
-        Ok(num_inserted_rows) => {
-            assert_eq!(num_rows_to_insert, num_inserted_rows);
-            Ok(())
-        }
-        Err(e) => {
-            error!("Failed to save records to db: {:?}", e);
-            Err(Error::internal_server_error())
-        }
+    let pool = &db::establish_connection().await?;
+    let mut num_inserted_rows: u64 = 0;
+    for key in keys_to_insert{
+        let result = match sqlx::query!(
+            r#"INSERT INTO keys (id, master_sae_id, slave_sae_id, size, content)VALUES ($1, $2, $3, $4, $5);"#,
+            key.id,
+            key.master_sae_id,
+            key.slave_sae_id,
+            key.size,
+            key.content,
+        ).execute(pool).await{
+            Ok(res) => res,
+            Err(e) => {
+                error!("Failed to save records to db: {:?}", e);
+                return Err(Error::internal_server_error());
+            }
+        };
+        num_inserted_rows += result.rows_affected();
     }
+    assert_eq!(num_rows_to_insert as u64, num_inserted_rows);
+    Ok(())
 }
 
-pub fn get_multiple_keys(
+pub async fn get_multiple_keys(
     key_ids: &[uuid::Uuid],
     master_sae_id: &str,
     slave_sae_id: &str,
 ) -> Result<Vec<Key>, Error> {
     let mut result: Vec<Key> = Vec::new();
 
-    let db_conn = &mut db::establish_connection()?;
+    let pool = &db::establish_connection().await?;
 
     for key_id in key_ids {
         result.push(retrieve_key_from_db(
             key_id,
             master_sae_id,
             slave_sae_id,
-            db_conn,
-        )?);
+            pool,
+        ).await?);
     }
 
     Ok(result)
 }
 
-fn retrieve_key_from_db(
+async fn retrieve_key_from_db(
     key_id: &uuid::Uuid,
     master_sae_id: &str,
     slave_sae_id: &str,
-    db_conn: &mut PgConnection,
+    pool: &PgPool,
 ) -> Result<Key, Error> {
-    use crate::schema::keys;
-
-    let num_keys_with_master_sae_id: i64 = match keys::table
-        .filter(keys::id.eq(key_id))
-        .filter(keys::master_sae_id.eq(master_sae_id))
-        .filter(keys::active.eq(true))
-        .count()
-        .get_result(db_conn)
-    {
+    let num_keys_with_master_sae_id = match sqlx::query!(
+        r#"SELECT count(*) as "count!"
+        FROM keys
+        WHERE 
+            id = $1 AND
+            master_sae_id = $2 AND
+            active = TRUE
+        ;"#,
+        key_id,
+        master_sae_id,
+    )
+    .fetch_one(pool)
+    .await{
         Ok(res) => res,
```

Comment:
*Suggestion:* If it aids code readability, consider retrieving the `count` field here.